torch/testing/_internal/common_device_type.py

```diff
@@ -428,6 +435,38 @@ def largeCUDATensorTest(size):
     return unittest.skipIf(not valid, "No CUDA or Has CUDA but GPU RAM is not large enough")
 
 
+def largeTensorTest(size):
+    """Skip test if the device has insufficient memory to run the test"""
+    if isinstance(size, str):
+        assert size.endswith("GB") or size.endswith("gb"), "only bytes or GB supported"
+        size = 1024 ** 3 * int(size[:-2])
+
+    def inner(fn):
+        @wraps(fn)
+        def dep_fn(self, *args, **kwargs):
+            if self.device_type == 'cuda':
+                valid = (torch.cuda.is_available() and
+                         torch.cuda.get_device_properties(0).total_memory >= size)
+            else:
+                if not HAS_PSUTIL:
+                    raise unittest.skip('Need psutil to determine if memory is sufficient')
+
+                # The sanitizers have significant memory overheads
+                if TEST_WITH_ASAN or TEST_WITH_TSAN or TEST_WITH_UBSAN:
+                    size *= 10
```

Comment:
The factor of 10 is a rough upper bound, based on these slides:
https://www.slideshare.net/sermp/sanitizer-cppcon-russia

Could also just skip entirely, if that's preferred.